cuda/base/thrust.cuh

```diff
@@ -0,0 +1,57 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright (c) 2017-2022, the Ginkgo authors
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#ifndef GKO_CUDA_BASE_THRUST_CUH_
+#define GKO_CUDA_BASE_THRUST_CUH_
+
+
+#include <thrust/execution_policy.h>
+#include <thrust/system/cuda/detail/execution_policy.h>
+
+
+namespace gko {
+namespace kernels {
+namespace cuda {
+
+
+inline auto thrust_policy(std::shared_ptr<const CudaExecutor> exec)
+{
+    return thrust::cuda::par.on(exec->get_stream());
```

Comment:
Is that part of the public thrust API? I couldn't find it in their online docs. If it's not part of the public API, do you know which versions support this?

Reply:
yes, the documentation is a bit bad in that regard. I am not sure whether the generic `thrust::par` would be equivalent. It should be supported by all versions of Thrust.

Reply:
Do you have any link to their docs for either of that? I still can't find anything on that.

Reply:
Sorry, I mixed up standardese with Thrust, it should be [thrust::device](https://thrust.github.io/doc/group__execution__policies_ga330d16be82b0518b6a37d026c2811e7e.html#ga330d16be82b0518b6a37d026c2811e7e), though more recent versions have thrust::par as well. The stream parameter is not documented anywhere though.

Reply:
I find that a bit troublesome, but I guess it works? At least I can find it in the headers directly.

Reply:
Thrust doesn't have a big team working on it, so they can't focus too much on documentation, sadly :(

Reply:
It is actually mentioned in the release notes of thrust 1.8.0 https://github.com/NVIDIA/thrust/releases/tag/1.8.0. Until it breaks, that is enough for me.